TorqueDataConnect/resources/js/TorqueDataConnectEdit.js

```diff
@@ -0,0 +1,224 @@
+/*@nomin*/
+function addEditButtonListeners () {
+    $(".torque-edit-button.paragraph").click(handleParagraphEdit);
+    $(".torque-edit-button.list").click(handleListEdit);
+
+    const [sheetName, , key] = $("#page-info")
+        .data("location")
+        .replace(".mwiki", "")
+        .split("/");
+    window.sheetName = sheetName;
+    window.key = key;
+}
+
+$(document).ready(() => {
+    addEditButtonListeners();
+});
+
+async function submitEdit (field, value, callback) {
+    const postData = {
+        newValues: JSON.stringify({ [field]: value }),
+        sheetName: window.sheetName,
+        key: window.key,
+        title: mw.config.values.wgTitle,
+    };
+    try {
+        // Gets CORS token for POST
+        const corsResult = await $.ajax({
+            type: "GET",
+            url: "/lfc/api.php?action=query&format=json&meta=tokens",
+            dataType: "json",
+        });
+
+        const token = encodeURIComponent(corsResult.query.tokens.csrftoken);
+        const actionName = "torquedataconnectsubmitedit";
+        const results = await $.ajax({
+            type: "POST",
+            url: `${mw.config.values.wgScriptPath}/api.php?action=${actionName}&format=json&token=${token}`,
+            data: postData,
+            dataType: "json",
+        });
+        $('.torque-wrapper').replaceWith(results.html);
+        addEditButtonListeners();
+    } catch (error) {
+        console.error(error);
+    }
+}
+
+const textArea = (v) => $(`<textarea name="" type="text">${v}</textarea>`);
+// Returns a jquery cancel and save button side-by-side
+const saveButtons = (fieldName, originalValue, onCancel, onSave) => {
+    const cancelBtn = $('<span class="torque-save-cancel">Cancel</span>');
+    cancelBtn.data("original", originalValue);
+    cancelBtn.click(onCancel);
+    const saveBtn = $('<span class="torque-save">Save</span>');
+    saveBtn.data("field", fieldName);
+    saveBtn.click(onSave);
+    return $(`<div id="${fieldName}" class="torque-save-wrapper"></div>`)
+        .append(cancelBtn)
+        .append(saveBtn);
+};
+const editButton = (type, field) => {
+    return $(
+        `<div class="torque-edit-button ${type}"><div class="torque-edit">edit</div></div>`
```

Comment:
'edit' here may mismatch the word chosen in the template.

Reply:
That is true. The formatting of the template must adhere to a specific style in order for this to work properly. Changes to the template are liable to break edit functionality.